prompting/miners/langchain_miner.py

```diff
@@ -0,0 +1,169 @@
+# The MIT License (MIT)
+# Copyright © 2024 Yuma Rao
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated
+# documentation files (the “Software”), to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
+# and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all copies or substantial portions of
+# the Software.
+
+# THE SOFTWARE IS PROVIDED “AS IS”, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+# THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+# OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+
+import time
+import os
+import bittensor as bt
+import argparse
+from starlette.types import Send
+from functools import partial
+from typing import Dict, Awaitable
+
+# Bittensor Miner Template:
+from prompting.base.prompting_miner import BaseStreamPromptingMiner
+from prompting.protocol import StreamPromptingSynapse
+
+# import base miner class which takes care of most of the boilerplate
+
+from prompting.miners.utils import OpenAIUtils
+
+from langchain.prompts import ChatPromptTemplate
+from langchain_core.output_parsers import StrOutputParser
+from langchain.chat_models import ChatOpenAI
+from dotenv import load_dotenv, find_dotenv
+from langchain_core.runnables.base import RunnableSequence
+from deprecated import deprecated
+
+@deprecated(version="2.4.1+", reason="Class is deprecated, use openai miner for reference on example miner.")
```

Comment:
maybe this is me being stupid, but is there a reason for adding a file with a deprecated class (if that file didn't exist originally?)

Reply:
It used to be called "openai miner" but then we wanted to support an openai miner that just purely queries the api without all the langchain fluff

Reply:
And we thought it best to offer this as a code example for agentic workflow